### ISSUE TITLE
[Wrong Usage] The Route Function /pages/all.page.route.ts should return a boolean or a plain JavaScript object, instead it returns `function Promise() { [native code] }`.

### ISSUE THREAD

**Issue by mike-jewell** [brillout/vite-plugin-ssr#425]
### Description

I am setting:

```
export const iKnowThePerformanceRisksOfAsyncRouteFunctions = true
```

I'm then exporting:

```
export default async (pageContext: PageContextBuiltIn) => {
```

The problem seems to be in vite-plugin-ssr\dist\cjs\shared\route\resolveRouteFunction.js:10:

```
await result;
```

It appears it should be:

```
result = await result;
```

**Comment by brillout** [comment]
Indeed it's a regression I mistakenly introduced in https://github.com/brillout/vite-plugin-ssr/commit/244d331842849586844cda36a751c86bf5ad2318#diff-5a0d6ac47b721604e70256b4000932c8909190641bba409adbca708d58aa7108L32-L36.

Let me fix this.

**Comment by brillout** [comment]
Fix released in `0.4.24`.
